sdk/search/Azure.Search.Documents/src/Models/FieldBase.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary>
+    /// Base field type for helper classes to more easily create a <see cref="SearchIndex"/>.
+    /// </summary>
+    public abstract class FieldBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FieldBase"/> class.
+        /// </summary>
+        /// <param name="name">The name of the field, which must be unique within the index or parent field.</param>
+        /// <param name="type">The data type of the field.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        public FieldBase(string name, DataType type)
+        {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Name = name;
+            Type = type;
+        }
+
+        /// <summary>
+        /// Gets the name of the field.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Ge the data type of the field.
+        /// </summary>
+        public DataType Type { get; }
+
+        /// <summary>
+        /// Persists class-specific properties into the given <see cref="SearchField"/>.
+        /// </summary>
+        /// <param name="field">The <see cref="SearchField"/> into which properties are persisted.</param>
+        protected abstract void Save(SearchField field);
+
+        /// <summary>
+        /// Casts a <see cref="FieldBase"/> or derivative to a <see cref="SearchField"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="FieldBase"/> or derivative to cast.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="value"/> is null.</exception>
+        public static implicit operator SearchField(FieldBase value)
```

Comment:
It doesn't have to. Each of the languages is approaching this with the same over all concept, but the language idioms play out differently. The only reason I'm even doing it this way is because we can't currently contain `SearchField` without a perf hit, but Pavel's recent changes to allow us to model structs opens that possibility to just containing a `SearchField` efficiently so we may not even need this by GA.

Reply:
To be clear, I'm talking about other .NET languages like F# which has Search customers but no support for calling implicit conversion operators _(other than an ugly `op_Implicit` manual call)_.  I think `Response.Value` is a fine alternative for them but I'm not seeing a good option for them to do this.  I think restructuring things a bit so it's just `ToSearchField()` would be easier approach to both this and the `Save` method naming.